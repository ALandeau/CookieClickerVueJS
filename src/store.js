import Vue from 'vue'
import VueCookie from 'vue-cookie'
import Vuex from 'vuex'
import { atoutStore } from "./atoutstore";

Vue.use(Vuex);

export default new Vuex.Store({
    state: {
        scoreNumber: VueCookie.get('score'),
        globalTimer: 0
    },
    getters: {
        scoreNumber: state => state.scoreNumber,
    },
    mutations: {
        incrementScoreNumber(state) {
            state.scoreNumber++;
            VueCookie.set('score', state.scoreNumber);
        },
        autoIncrementScoreNumber(state) {
            setInterval(function () {
                state.globalTimer = +state.atoutStore.grandmaNumber + (+state.atoutStore.cursorNumber / 10);
                state.scoreNumber = +state.scoreNumber + +state.globalTimer;
                VueCookie.set('score', state.scoreNumber);
            }, 1000);
        },
        grandmaInteract(state) {
            let grandmaPrice = state.atoutStore.grandmaPrice;
            state.scoreNumber = state.scoreNumber - grandmaPrice;
            state.atoutStore.grandmaNumber++;
            grandmaPrice = +grandmaPrice * (((state.atoutStore.grandmaNumber / 10)) + 1);
            grandmaPrice = Math.round(grandmaPrice);
            state.atoutStore.grandmaPrice = grandmaPrice;
            VueCookie.set('grandma', state.atoutStore.grandmaNumber)
        },
        cursorInteract(state) {
            let cursorPrice = state.atoutStore.cursorPrice;
            state.scoreNumber = state.scoreNumber - cursorPrice;
            state.atoutStore.cursorNumber++;
            cursorPrice = +cursorPrice * (((state.atoutStore.cursorNumber / 10)) + 1);
            cursorPrice = Math.round(cursorPrice);
            state.atoutStore.cursorPrice = cursorPrice;
            VueCookie.set('cursor', state.atoutStore.cursorNumber)
        }
    },
    modules: {
        atoutStore,
    },
});