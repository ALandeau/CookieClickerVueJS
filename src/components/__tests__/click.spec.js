import { mount, createLocalVue } from "@vue/test-utils";
import Vuex from "vuex";
import Click from "@/components/clickCookie.vue";
import store from "@/store";

describe("clickCookie.vue", () => {
  const localVue = createLocalVue();
  localVue.use(Vuex);

  const wrapper = mount(Click, {
    store,
    localVue
  });

  it("renders the correct markup", () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("button click should increment the count", () => {
    expect(store.state.scoreNumber).toBe(0);
    const counter = wrapper.find("#score .input-block");
    counter.trigger("click");
    expect(store.state.scoreNumber).toBe(1);
  });
});
