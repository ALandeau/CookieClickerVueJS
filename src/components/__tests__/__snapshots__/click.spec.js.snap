// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`clickCookie.vue renders the correct markup 1`] = `
<div id="score">
  <h2>NaN</h2>
  <div class="input-block">
    <img src="../assets/images/perfectCookie.png" alt="perfect cookie">
  </div>
  <button>Reset Value</button>
</div>
`;
